components/brave_player/resources/playerEmbed.tsx

```diff
@@ -0,0 +1,61 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import * as React from 'react'
+import { render } from 'react-dom'
+import styled from 'styled-components'
+
+// For now we only supports youtube.
+// Usages: EmbedURL`youtube${videoId}`
+export function EmbedURL (source: string, videoId: string) {
+  if (source === 'youtube')
+    return `https://www.youtube-nocookie.com/embed/${videoId}?iv_load_policy=1&autoplay=1&rel=0&modestbranding=1`
+
+  return ''
+}
+
+const EmbedContainer = styled.div`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  height: 100%;
+`
+
+const StyledFrame = styled.iframe`
+  width: 80%;
+  height: 80%;
+  border: none;
+`
+
+export function getFrameSrc (url: URL) {
+  let paths = url.pathname.split('/')
+  if (paths.length < 3) {
+    return ''
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [_, service, videoId] = paths
+
+  return EmbedURL(service, decodeURIComponent(videoId))
+}
+
+function PlayerEmbed () {
+  return (
+    <EmbedContainer>
+      <StyledFrame
+        src={getFrameSrc(new URL(window.location.href))}
```

Comment:
Optional:

I think `getFrameSrc` only needs the pathname so you could update it to take a string and just pass the `pathname` in here, if you don't want to do the `new URL`

Alternatively, you could change the param type to `readonly URL` and directly pass in window.location:

```tsx
function getFrameSrc(url: Readonly<URL>) {}
getFrameSrc(window.location)
```

Reply:
Nice. I tried the second one too but interesetingly, `window.location` seems to be not a `URL` type, but readonly "Location" type. So changed to take `pathname: string` 👍